aws/src/main/java/org/apache/iceberg/aws/AwsClientFactories.java

```diff
@@ -143,31 +134,12 @@ public DynamoDbClient dynamo() {
 
     @Override
     public void initialize(Map<String, String> properties) {
+      this.awsProperties = new AwsProperties(properties);
       this.glueEndpoint = properties.get(AwsProperties.GLUE_CATALOG_ENDPOINT);
```

Comment:
given we already have awsProperties stored, I think we can remove all the other variables that are duplicated in `awsProperties`

Reply:
Thank you for your suggestions. I moved all endpoint fields: `glueEndpoint`, `s3Endpoint`, and `dynamodbEndpoint` to `AwsProperties` so that we do not have to parse them separately in every client factory.